fineract-accounting/src/main/java/org/apache/fineract/accounting/producttoaccountmapping/service/ProductToGLAccountMappingHelper.java

```diff
@@ -455,4 +552,37 @@ public void deleteProductToGLAccountMapping(final Long loanProductId, final Port
             this.accountMappingRepository.deleteAllInBatch(productToGLAccountMappings);
         }
     }
+
+    public void validateChargeOffMappingsInDatabase(final List<JsonObject> mappings) {
+        final List<ApiParameterError> validationErrors = new ArrayList<>();
+
+        for (JsonObject jsonObject : mappings) {
+            final Long expenseGlAccountId = this.fromApiJsonHelper
+                    .extractLongNamed(LoanProductAccountingParams.EXPENSE_GL_ACCOUNT_ID.getValue(), jsonObject);
+            final Long chargeOffReasonCodeValueId = this.fromApiJsonHelper
+                    .extractLongNamed(LoanProductAccountingParams.CHARGE_OFF_REASON_CODE_VALUE_ID.getValue(), jsonObject);
+
+            // Validation: chargeOffReasonCodeValueId must exist in the database
+            CodeValue codeValue = this.codeValueRepository.findByCodeNameAndId("ChargeOffReasons", chargeOffReasonCodeValueId);
+            if (codeValue == null) {
+                validationErrors.add(ApiParameterError.parameterError("validation.msg.chargeoffreason.invalid",
+                        "Charge-off reason with ID " + chargeOffReasonCodeValueId + " does not exist",
+                        LoanProductAccountingParams.CHARGE_OFF_REASONS_TO_EXPENSE.getValue()));
+            }
+
+            // Validation: expenseGLAccountId must exist as a valid Expense GL account
+            final Optional<GLAccount> glAccount = accountRepository.findById(expenseGlAccountId);
```

Comment:
This validation is checking whether the gl account does exist, but it does not check whether this gl account is expense type, right?

Reply:
Yes

Reply:
Will update

Reply:
Updated